tag_dedupe/dedupe.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/python
+
+import sys
+import subprocess
+import os
+import argparse
+
+AUTH_FILE_PATH_LOCAL = "/auth.json"
+
+def dedupe(input_tag, existing_tags):
+	if input_tag in existing_tags:
+		sys.exit("Tag already exists in remote repository! Exiting build.")
+	print "Tag does not exist in remote repository! Continuing with build."
+
+
+# Since gcloud auth information doesn't transfer automatically into a new container in prod, 
+# we use a placeholder service account so we can make authenticated calls to the gcloud API.
+def login(AUTH_FILE_PATH_REMOTE):
+	#######################################################################################
+	# TODO: once we can make unauthenticated gsutil calls in prod, 
+	# uncomment this and remove the temporary auth.json file from the docker image.
```

Comment:
this is commented out because unfortunately it does not work in production with the current version of `gsutil`. the objective here is to copy down the auth file for a service account from cloud storage, so we can log in with that account and make gcloud calls; the issue with that is that the `gsutil` version in production does not allow for ANY commands to be issued, regardless of whether they require credentials, without being logged in. so here i'm just defaulting to using the auth file that's been copied into the container when the image is built.